src/Sema.zig

```diff
@@ -9350,6 +9337,150 @@ fn fieldPtr(
     return mod.fail(&block.base, src, "type '{}' does not support field access", .{object_ty});
 }
 
+fn fieldCallBind(
+    sema: *Sema,
+    block: *Scope.Block,
+    src: LazySrcLoc,
+    raw_ptr: Air.Inst.Ref,
+    field_name: []const u8,
+    field_name_src: LazySrcLoc,
+) CompileError!Air.Inst.Ref {
+    // When editing this function, note that there is corresponding logic to be edited
+    // in `fieldVal`. This function takes a pointer and returns a pointer.
```

Comment:
Oof is this the 3rd copy of field lookup logic? I'm OK with this for now, but we'll probably want to think about how to DRY this up a bit in the future.

Reply:
I agree, I couldn't see an easy way to compress it since I'm new to this code but it definitely should be done.